src/kakarot/library.cairo

```diff
@@ -218,10 +220,46 @@ namespace Kakarot {
     }(evm_contract_address: felt) -> (starknet_contract_address: felt) {
         alloc_locals;
         let (starknet_contract_address) = Starknet.deploy(evm_contract_address);
-
         return (starknet_contract_address=starknet_contract_address);
     }
 
+    // @notice Return the class hash of the account implementation
+    // @return account_contract_class_hash The class hash of the account implementation
+    func get_account_contract_class_hash{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+    }() -> (account_contract_class_hash: felt) {
+        let (account_contract_class_hash) = Kakarot_account_contract_class_hash.read();
+        return (account_contract_class_hash,);
+    }
+
+    // @notice Registers an account by writing its EVM address to the Starknet address mapping.
+    // @dev Called by the account contract upon initialization.
+    // @param evm_address The EVM address of the account.
+    func register_account{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        evm_address: felt
+    ) {
+        alloc_locals;
+
+        let (existing_address) = Kakarot_evm_to_starknet_address.read(evm_address);
+        with_attr error_message("Kakarot: account already registered") {
+            assert existing_address = 0;
+        }
+
+        let (local caller_address: felt) = get_caller_address();
+        let starknet_address = Account.compute_starknet_address(evm_address);
+        local starknet_address = starknet_address;
+        let (local kakarot_address: felt) = get_contract_address();
+
+        with_attr error_message(
+                "Kakarot: register account address mismatch {starknet_address} != {caller_address} != {kakarot_address} ") {
```

Comment:
maybe an error like "caller should be corresponding starknet address" ?

Reply:
sorry that's from a debugging session where I could only use the error message to have values...

Reply:
`                "Kakarot: Caller should be {starknet_address}, got {caller_address}") {`